File: py/src/langchain/first_chain.py
from langchain_aws import BedrockLLM as Bedrock
from langchain_core.prompts import PromptTemplate
import boto3

AWS_REGION = "us-west-2"

bedrock = boto3.client(service_name="bedrock-runtime", region_name=AWS_REGION)

model = Bedrock(model_id="amazon.titan-text-express-v1", client=bedrock)


def invoke_model():
    response = model.invoke("What is the highest mountain in the world?")
    print(response)


def first_chain():
    # Use a plain PromptTemplate with a text LLM (Titan Text Express)
    prompt = PromptTemplate.from_template(
        "Write a short, compelling product description for: {product_name}"
    )
    chain = prompt | model

    response = chain.invoke({"product_name": "bicycle"})
    print(response)


first_chain()

File: ts/src/langchain/FirstChain.ts
import { BedrockChat } from '@langchain/community/chat_models/bedrock';
import { ChatPromptTemplate } from '@langchain/core/prompts'

const AWS_REGION = 'us-west-2'

const model = new BedrockChat({
    model: 'amazon.titan-text-express-v1',
    region: AWS_REGION
});

async function invokeModel() {
    const response = await model.invoke("What is the highest mountain in the world?");

    console.log(response)
}

async function firstChain() {
    const prompt = ChatPromptTemplate.fromMessages([
        ['system', 'Write a short description for the product provided by the user'],
        ['human', '{product_name}']
    ])

    const chain = prompt.pipe(model);

    const response = await chain.invoke({
        product_name: 'bicycle'
    })
    console.log(response)
}

firstChain()

